src/resolvers/message.ts

```diff
@@ -39,23 +39,56 @@ const resolver: Resolvers = {
           };
         }
 
-        const authUsers = [...users, auth.userId];
-        const memberships = await membershipModel.findAll({
-          attributes: [
-            'channelId',
-            [sequelize.fn('count', sequelize.fn('DISTINCT', sequelize.col('userId'))), 'memberCnt'],
-          ],
-          where: {
-            userId: authUsers,
-          },
+        const authUsers = [...new Set([...users, auth.userId])];
+        const channels = await channelModel.findAll({
           group: ['channelId'],
+          having: sequelize.where(
+            sequelize.fn('COUNT', sequelize.col('channelId')),
+            { [Op.in]: [authUsers.length] },
+          ),
+          include: [
+            {
+              model: membershipModel,
+              as: 'memberships',
+              where: {
+                userId: authUsers,
+              },
+              attributes: [
+                'channelId',
+              ],
+            },
+          ],
           raw: true,
         });
 
-        const retrievedChannel =
-          memberships.find((membership) => membership.memberCnt === authUsers.length);
+        let retrievedChannelId: string;
+        const channelPromises = [];
+
+        channels.forEach((channel, index) => {
+          channelPromises[index++] = channelModel.findOne({
```

Comment:
I think this query looks a bit redundant which I've concerned previously. Query all the channels with retried channel ids may result in poor performance. If this is our best, I think it'd be good to consider on `group key` as we've previously discussed.

Reply:
Ok. I understand your point. At first glance, looping all filtered channels just makes a couple of more queries. But after listening to your opinion, it seems that the above code can possibly raise a serious performance issue when dealing with large data.
If you already decided to make a group key instead, you can close this PR.

By the way, thank you so much for the review :)

Reply:
Thanks for trying tough it was much nice to visualize the complexity on the approach.

Could you also try to use sequelize raw query on prev PR which we could not make the distinct?

Reply:
Does your last sentence mean improving this #67 ( #66 too ) PR by using the raw sql query with sequelize before we choose to make group keys? As we haven't found a good proper solution for finding the existing channel.
Or did you mention something else?

Reply:
Oh I thought you still have solution on solving current issue with the raw query but if it's not, lets change our plan for using a group key.

Reply:
@hyochan 
OK!
I'll give it a shot with raw sql query!

Tell you the result till tomorrow.